tests/test_distributions.py

```diff
@@ -546,3 +546,39 @@ def test_extra_parameters(tf_seed, distribution_extra_parameters):
     if distribution_name not in ["Flat", "HalfFlat"]:
         # Test that a sample can be drawn using the alternative extra parameters values
         dist.sample()
+
+
+@pytest.fixture(scope="function", ids=str)
```

Comment:
Ok, I see that we have to do many mixture specific tests so it looks like they should be moved to their own test script. Could you also test the prior and posterior predictive sampling, and also check if a multidimensional `p` and a batch of distributions work too (something like the `multidim_parameters` that we test in other distributions).

Reply:
Hey @lucianopaz , I've implemented some more tests for the mixture class, but some issues came up for the prior/posterior checks regarding shapes which I need to dig into.

I'm not sure though if this is a bug on my side or on TFP's side, since I there's similar problems if I implement the mixture in plain TFP (here's a [colab](https://colab.research.google.com/drive/1XdzhIUANslwKaAYtHUbEBvxUaAFyWFH9?usp=sharing) showing the issue), so any tips would be appreciated.

Reply:
It looks like a potentially painful issue that vmap runs into when it draws a batch of labels and tries to index into the mixture components list. Could you try passing `use_static_graph=True` to see if that solves the problem?

Reply:
@junpenglao, could you take a look at the error that is popping out in the colab? Maybe it's worth opening an issue on tfp?

Reply:
@junpenglao might you have a look a this?

Reply:
@dirmeier can you also please consider adding tests with `xla=True`. (I'm not sure if we should flag it as `xfail` though)

Reply:
Sure :)

Reply:
@dirmeier are there any issues on pymc4 side still? Disabling dynamic sampling of mixture elements as @lucianopaz recommends, seems to resolve the issues with sampling and the second issue is more related to the tfp coroutines (maybe opening a separate issue in tfp?).

Reply:
Hi, sorry for the delay. If one sest `use_static_graph` sample drom prior/posterior predictive distributions works. But it would need to be hardcoded in `_init_distribution` as you cant set it via `kwargs`.

Reply:
Maybe we can add an additional logic in prior/posterior predictive to modify mixture distribution instance attribute (or check whenever it is `use_static_graph=True`) after the model is evaluated. Or maybe overriding `proceed_distribution` to modify the instance attribute.

Reply:
I don't have a preference here. I think using a static graph per default when instantiating the mixture is probably the easiest and cleanest option. @lucianopaz what do you think on this?

Reply:
ping @lucianopaz

Reply:
@dirmeier, sorry for the delay, I kind of missed what was going on here. Did you finally push some version where you used the static graph? Did it solve the problem?